File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let insertIdx = 1, uniqueIdx = 1;
    while (nums[insertIdx-1] !== nums[insertIdx] && insertIdx < nums.length) insertIdx++;
    while (uniqueIdx < nums.length && insertIdx < nums.length) {
        while (nums[uniqueIdx] <= nums[insertIdx-1]  && uniqueIdx < nums.length) uniqueIdx++;
        if (uniqueIdx < nums.length) {
            [nums[insertIdx], nums[uniqueIdx]] = [nums[uniqueIdx], nums[insertIdx]];
            insertIdx++;
        }
    }
    return insertIdx
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.swift
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        var lo = 0, hi = 1;

        while hi < nums.count {
            
            if nums[lo] == nums[hi] {

            } else {
                lo += 1;
                nums.swapAt(lo, hi);
            }
            hi += 1;
        }
        return lo + 1;
    }
}